lib/matplotlib/dates.py

```diff
@@ -1175,22 +1181,28 @@ def __call__(self):
         return self.tick_values(dmin, dmax)
 
     def tick_values(self, vmin, vmax):
-        delta = relativedelta(vmax, vmin)
+        if self.mult_base:
+            # start and stop need to be multiples of the interval
```

Comment:
I'm not sure I follow this.  Why do start and stop need to be multiples of the intervals?  This either needs more comment, or cleaning up?

Reply:
More comments probably? So, usually creating ticks on multiples of the interval is done by adjusting the "byranges". So for example with frequency =SECONDLY and interval = 15, the AutoDateLocator makes `byseconds = [0, 15, 30, 45]`. 

But there is no "byrange" for years. Rrule can create a reoccurrence every 10 years (for example). But you cannot specify that you want that to be on every full decade only. It will be every 10 years starting with the start date. So to get multiples of the interval, start needs to be a multiple of the interval. 

This is why AutoDateLocator currently returns an RRuleLocator for YEARLY when `.interval_multiples = False`, but a YearLocator for when it is true.

Stop can technically be a value greater than the last desired tick. 

The comment is not very clear, I agree. It should probably be something like "start needs to be a multiple of the interval for ticks on interval multiples when freq == YEARLY"

Reply:
> But there is no "byrange" for years. Rrule can create a reoccurrence every 10 years (for example). But you cannot specify that you want that to be on every full decade only. It will be every 10 years starting with the start date. So to get multiples of the interval, start needs to be a multiple of the interval.

Probably something like this in the comments?   That makes sense...